reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerConfig.java

```diff
@@ -998,6 +1033,20 @@ public void handlerAdded(ChannelHandlerContext ctx) {
 			}
 			pipeline.remove(NettyPipeline.HttpTrafficHandler);
 			pipeline.remove(NettyPipeline.ReactiveBridge);
+
+			if (idleTimeout != null) {
+				IdleTimeoutHandler.removeIdleTimeoutHandler(pipeline);
+				IdleTimeoutHandler.addIdleTimeoutServerHandler(
+						pipeline,
+						idleTimeout,
+						new Http2ConnectionLiveness(
+								upgrader.http2FrameCodec,
+								http2SettingsSpec != null ? http2SettingsSpec.pingAckTimeout() : null,
+								http2SettingsSpec != null ? http2SettingsSpec.pingScheduleInterval() : null,
+								http2SettingsSpec != null ? http2SettingsSpec.pingAckDropThreshold() : null
+						)
+				);
+			}
```

Comment:
In `H2CleartextCodec`, when upgrading from `HTTP/1.1` to the `H2C` protocol, the `IdleTimeoutHandler` is changed to be based on `Http2ConnectionLiveness`.